test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +95,11 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (commandType.isMutating()) {
+            assertEquals(addressBook, saveFile.load());
+        } else {
+            assertEquals(new AddressBook(), saveFile.load());
```

Comment:
:+1: good job on covering both the conditions!
However, you can include a check to see if in-memory data is still intact :sweat_smile: